### ISSUE TITLE
Reopen previous ZED state, while opening a new file

### ISSUE THREAD

**Issue by krstp** [zed-industries/community#1338]
### Check for existing issues

- [X] Completed

### Describe the feature

- Currently ZED remembers the previous UI (tabs etc.) state when opening ZED (call it `ABC`).
- However, when opening fresh ZED by opening a different file (call it `D`), the previous ZED state (all open windows, project - `ABC`) is not opened as if all the windows/tabs were lost... not being able to bring back the previous project/tabs back.... only seeing the `D` file edit.
- Now is an interesting plot... When quitting that newly opened window (`D`) with this different file edit (`D`) and reopening ZED fresh, it brings back not the previous last remembered window (`D`), but actually brings back the previous state (`ABC`) with all the project/tabs open, but now forgetting that previous the single file edit (`D`).

I hope it clarifies.

### If applicable, add mockups / screenshots to help present your vision of the feature

- Initial project `ABC`:
&nbsp; &nbsp; ![ABC](https://user-images.githubusercontent.com/4711498/229298657-6387d5ea-a1a9-4203-bb57-41d8accc341f.png)
- close app with `ABC` state
- open different `D` file: 
![D](https://user-images.githubusercontent.com/4711498/229298676-3f352d66-1cd6-43fe-82eb-940071afe886.png)
    - `ABC` does not get opened, but one would think it should as well
- close app with `D` state
- reopen app
   - `ABC` state reopens; `D` state gets lost ethernally. 
   ![ABC](https://user-images.githubusercontent.com/4711498/229298657-6387d5ea-a1a9-4203-bb57-41d8accc341f.png)

**Comment by krstp** [comment]
To add to it, ZED generally does not remember the state of opened windows, such as multiple project windows or singled out files in stand-alone windows upon close are not remembered (which is really annoying). When ZED is reopened it will only open the the lastly defined/opened project window, which is kind of frustrating. 😮‍💨 

Samples:
- working projects sample: `back-left`: large project window, `top-right`: a file edit with terminal opened by default (other issue #1398), `right-bottom`: new project window (the most recently opened ZED project)
![Working_projects](https://user-images.githubusercontent.com/4711498/230622488-d457b777-04dd-46b1-9aef-2e967178f2d5.png)

- sample after ZED close and reopen: notice `back-left` (main project) window and `top-right` edit-file disappear, while new `right-bottom` project is reopened.
![Reopened](https://user-images.githubusercontent.com/4711498/230623054-22dc2673-40e6-4118-a5c3-72bd45dd3198.png)

In summary: ZED should remember the state of all opened projects and windows.
